### PR TITLE
Fix uninitialized variables and add checks for null pointers

### PR DESCRIPTION
**Pull request by aobolensk**

### CODE REVIEW

File: _studio/mfx_lib/decode/mjpeg/src/mfx_mjpeg_dec_decode.cpp [hunk 1]
```diff
@@ -664,7 +664,7 @@ mfxStatus VideoDECODEMJPEG::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1
     // It can be useful to run threads right after first frame receive
     MFX_SAFE_CALL(DecodeFrameCheck(bs, surface_work, surface_out));
 
-    UMC::FrameData *dst;
+    UMC::FrameData *dst = NULL;
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/mjpeg/src/mfx_mjpeg_dec_decode.cpp:667]: ```nullptr```

File: _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp [hunk 2]
```diff
@@ -3090,6 +3090,7 @@ mfxStatus VideoVPPHW::VppFrameCheck(
             }
         }
 
+        MFX_CHECK_NULL_PTR1(pWorkOutSurf);
```
Comment (mgonchar) [inline _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp:3093]: Why it is not inside ```AssignTask```?

File: _studio/mfx_lib/decode/mjpeg/src/mfx_mjpeg_dec_decode.cpp (hunk 1 above)
Comment (aobolensk) [reply _studio/mfx_lib/decode/mjpeg/src/mfx_mjpeg_dec_decode.cpp]: Done

File: _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp (hunk 2 above)
Comment (aobolensk) [reply _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp]: Moved this check inside `AssignTask` function

**Pull request merged**